### PR TITLE
Spec for new disk inventory structure in etcd

### PR DESCRIPTION
**Pull request by nnDarshan**
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@brainfunked @nthomas-redhat @shtripat @GowthamShanmugam Please have a look

File: specs/disk_inventory_etcd_structure.adoc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+== Proposed change
+
+The disk inventory details will basically stored under 2 sections as follows:
+
+1. /nodes/<node-id>/disks/
+
+* In this location we store all the physical disks available on the node
+with a unique key which is a combination of manufacturer-id+vendor-id+
+device-serial-number this would be unique across tendrl. And this information
+should be available from hwinfo tool.
```
Comment (r0h4n) [inline specs/disk_inventory_etcd_structure.adoc:26]: provide an example of such output from hwinfo, and also add the same in the disk sync code

File: specs/disk_inventory_etcd_structure.adoc [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+* under each disk here we have to store all the details about the physical
+  device available via hwinfo tool.
+
+* also the following details have to be stored under each disk(if applicable):
+       - partition details under it
+       - lvm details under
+       - raid specific details
+
+2. /nodes/<node-id>/block-devices/
```
Comment (r0h4n) [inline specs/disk_inventory_etcd_structure.adoc:36]: id prefer it to be "/nodes/<node-id>/block_devices/"

We dont use hyphen in "list" attributes of objects

File: specs/disk_inventory_etcd_structure.adoc [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+         free block device from /free to /used.
+      2. Also the links between block devices and disks have to be kept up to date,
+         so that even if block-device name changes after reboot. it should properly
+         be reflected in the disk inventory.
+
+=== Alternatives
+
+None
+
+=== Data model impact
```
Comment (r0h4n) [inline specs/disk_inventory_etcd_structure.adoc:69]: provide attributes of the objects discussed in this spec
-  "tendrl.objects.Disk" 
- "tendrl.objects.BlockDevice"

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: specs/disk_inventory_etcd_structure.adoc (hunk 1 above)
Comment (GowthamShanmugam) [reply specs/disk_inventory_etcd_structure.adoc:26]: @r0h4n @brainfunked VM disk does not have serial-number, we can use this key combination 

and hwifo not giving manufacturing_id also,   
output of hwifo from VM disk:
  Unique ID: WiMi.Fxp0d3BezAE
  Parent ID: vkLc.+FFPFBVXZu6
  SysFS ID: /class/block/vdh
  SysFS BusID: virtio9
  SysFS Device Link: /devices/pci0000:00/0000:00:0e.0/virtio9
  Hardware Class: disk
  Model: "Disk"
  Driver: "virtio-pci", "virtio_blk"
  Driver Modules: "virtio_pci", "virtio_blk"
  Device File: /dev/vdh
  Device Files: /dev/vdh, /dev/disk/by-id/virtio-78065dbf-92db-4574-8, /dev/disk/by-path/virtio-pci-0000:00:0e.0
  Device Number: block 252:112-252:127
  Geometry (Logical): CHS 31207/16/63
  Size: 31457280 sectors a 512 bytes
  Capacity: 15 GB (16106127360 bytes)
  Config Status: cfg=new, avail=yes, need=no, active=unknown
  Attached to: #27 (Storage controller)
 
it is not giving vendor details also for vm disks, all three ids not present in VM disk

Comment (r0h4n) [reply specs/disk_inventory_etcd_structure.adoc:26]: whats the update on vm disks>

Comment (GowthamShanmugam) [reply specs/disk_inventory_etcd_structure.adoc:26]: We can use field from hwifo  
Device Files: /dev/sda, /dev/disk/by-id/ata-SAMSUNG_MZ7TE512HMHP-000L1_S1GJNSAG400778, /dev/disk/by-id/wwn-0x4d30445853885002

if you see here by-id is never change even disk is removed and connected again, problem is it gives two by-id
 In VM it give one by-id those format are different 
Device Files: /dev/vdh, /dev/disk/by-id/virtio-78065dbf-92db-4574-8, /dev/disk/by-path/virtio-pci-0000:00:0e.0

so i don't  know what to do

**Pull request closed without merging**